crates/ruff_python_formatter/src/expression/expr_yield.rs

```diff
@@ -49,3 +59,58 @@ impl NeedsParentheses for ExprYield {
         }
     }
 }
+
+impl<'a> From<&'a ExprYield> for AnyExpressionYield<'a> {
+    fn from(value: &'a ExprYield) -> Self {
+        AnyExpressionYield::Yield(value)
+    }
+}
+
+impl<'a> From<&'a ExprYieldFrom> for AnyExpressionYield<'a> {
+    fn from(value: &'a ExprYieldFrom) -> Self {
+        AnyExpressionYield::YieldFrom(value)
+    }
+}
+
+impl<'a> From<&AnyExpressionYield<'a>> for AnyNodeRef<'a> {
+    fn from(value: &AnyExpressionYield<'a>) -> Self {
+        match value {
+            AnyExpressionYield::Yield(yld) => AnyNodeRef::ExprYield(yld),
+            AnyExpressionYield::YieldFrom(yld) => AnyNodeRef::ExprYieldFrom(yld),
+        }
+    }
+}
+
+impl Format<PyFormatContext<'_>> for AnyExpressionYield<'_> {
+    fn fmt(&self, f: &mut Formatter<PyFormatContext<'_>>) -> FormatResult<()> {
+        let expr = if self.is_yield_from() {
```

Comment:
nit: rename this to something like `keyword`, otherwise it's confusing since the value represented is not an expression, while `val` is an expression